### ISSUE TITLE
Clicking the "Advanced" cog should set focus on the block tab

### ISSUE THREAD

**Issue by jasmussen** [WordPress/gutenberg#2304]
Current behavior:

- Select a block
- Click the cog and the block sidebar opens
- Press tab and you're taken to the next block

Desired behavior:

- Select a block
- Click the cog and the block sidebar opens, and focus is set on the first interactable element in the block sidebar
- Press tab and you're taken to the next interactable element in the block sidebar

In addition to setting focus on the sidebar, we should visually flash the sidebar to show where focus is now. Something like this, perhaps: https://codepen.io/joen/pen/Nvjror

CC: @afercia

**Comment by afercia** [comment]
@jasmussen thanks. See also:
"Keyboard navigation from blocks to the sidebar and back to blocks" #1182
where we were wondering about skip links

and https://github.com/WordPress/gutenberg/issues/1608#issuecomment-313148586

The argument there is that moving focus programmatically can be an assumption about one specific workflow, while users might want to just explore content or visually check the sidebar settings or any other workflow users will able to find 🙂 

A similar feature is implemented in the Media modal: when selecting an attachment, focus gets moved to the details sidebar. This actually forces users to tab dozens (potentially hundreds) of times to get back to the attachment grid if they want to select multiple attachments or unselect the previously selected one or perform any other action outside the sidebar.
Over time, we've received lots of complaints about this  feature and I'd really like to don't do the same mistake again 🙂 

Also, if we're going to implement keyboard shortcuts _à la_ Slack (and landmarks) to jump through the main UI sections as proposed in https://github.com/WordPress/gutenberg/issues/467#issuecomment-320719586 maybe moving focus is not a concern any longer?

**Comment by jasmussen** [comment]
Closing this in favor of #1182
